src/codegen/codegen_llvm.cpp

```diff
@@ -320,67 +372,299 @@ void CodeGen_LLVM::visit(const Rem* op) {
+
+    for (size_t i = 2; i < op->operands.size(); i++) {
+      Builder->CreateStore(value, allocaValue);
+      auto minValue = Builder->CreateLoad(allocaValue);
+      auto c = codegen(op->operands[i]);
+      icmp = (op->type.isInt()) ? Builder->CreateICmpSLT(minValue, c)
+                                : Builder->CreateICmpULT(minValue, c);
+      value = Builder->CreateSelect(icmp, minValue, c);
+    }
+  }
 }
 
 void CodeGen_LLVM::visit(const Max* op) {
   auto _ = CodeGen_LLVM::IndentHelper(this, "Max");
-  throw logic_error("Not Implemented for Max.");
+  taco_iassert(op->operands.size() >= 2);
+
+  auto a = codegen(op->operands[0]);
+  auto b = codegen(op->operands[1]);
+
+  if (op->type.isFloat()) {
+    // LLVM's minnum intrinsic only does binary ops
+    value = Builder->CreateMaxNum(a, b);
+    for (size_t i = 2; i < op->operands.size(); i++) {
+      value = Builder->CreateMaxNum(value, codegen(op->operands[i]));
+    }
+  } else {
+    llvm::Value* allocaValue = nullptr;
+    if (op->operands.size() > 2)
+      allocaValue = Builder->CreateAlloca(a->getType());
+
+    llvm::Value* icmp =
+        (op->type.isInt()) ? Builder->CreateICmpSGT(a, b) : Builder->CreateICmpUGT(a, b);
+    value = Builder->CreateSelect(icmp, a, b);
+
+    for (size_t i = 2; i < op->operands.size(); i++) {
+      Builder->CreateStore(value, allocaValue);
+      auto maxValue = Builder->CreateLoad(allocaValue);
+      auto c = codegen(op->operands[i]);
+      icmp = (op->type.isInt()) ? Builder->CreateICmpSGT(maxValue, c)
+                                : Builder->CreateICmpUGT(maxValue, c);
+      value = Builder->CreateSelect(icmp, maxValue, c);
+    }
+  }
 }
 
 void CodeGen_LLVM::visit(const BitAnd* op) {
   auto _ = CodeGen_LLVM::IndentHelper(this, "BitAnd");
-  throw logic_error("Not Implemented for BitAnd.");
+  value = Builder->CreateAnd(codegen(op->a), codegen(op->b));
 }
 
 void CodeGen_LLVM::visit(const BitOr* op) {
   auto _ = CodeGen_LLVM::IndentHelper(this, "BitOr");
-  throw logic_error("Not Implemented for BitOr.");
+  value = Builder->CreateOr(codegen(op->a), codegen(op->b));
 }
 
 void CodeGen_LLVM::visit(const Eq* op) {
   auto _ = CodeGen_LLVM::IndentHelper(this, "Eq");
-  throw logic_error("Not Implemented for Eq.");
+  auto* a = codegen(op->a);
+  auto* b = codegen(op->b);
+
+  if (op->type.isFloat()) {
+    if (auto* FPa = llvm::dyn_cast<llvm::ConstantFP>(a)) {
+      if (auto* FPb = llvm::dyn_cast<llvm::ConstantFP>(b)) {
+        if (FPa->isNaN() || FPb->isNaN())
+          value = Builder->CreateFCmpUEQ(a, b);  // UEQ means that either operand may be a QNAN.
+        else
+          value = Builder->CreateFCmpOEQ(a, b);  // OEQ means that neither operand is a QNAN
+      } else {
+        value = Builder->CreateFCmpUEQ(a, b);
+      }
+    } else {
+      value = Builder->CreateFCmpUEQ(a, b);
+    }
+  } else {
+    taco_iassert(op->type.isInt() || op->type.isUInt()) << "op->type is " << op->type;
```

Comment:
`op->type` can be boolean here as well
```suggestion
    taco_iassert(op->type.isInt() || op->type.isUInt() || op->type.isBool()) << "op->type is " << op->type;
```